app/scenes/Document/components/PublicBreadcrumb.js

```diff
@@ -0,0 +1,84 @@
+// @flow
+import { observer } from "mobx-react";
+import * as React from "react";
+import { Crumb, Slash } from "components/Breadcrumb";
+import Flex from "components/Flex";
+import BreadcrumbMenu from "menus/BreadcrumbMenu";
+import { type NavigationNode } from "types";
+
+type Props = {|
+  documentId: string,
+  shareId: string,
+  sharedTree: NavigationNode,
+  children?: React.Node,
+|};
+
+function pathToDocument(sharedTree: NavigationNode, documentId: string) {
+  let path = [];
+  const traveler = (nodes, previousPath) => {
+    nodes.forEach((childNode) => {
+      const newPath = [...previousPath, childNode];
+      if (childNode.id === documentId) {
+        path = newPath;
+        return;
+      }
+      return traveler(childNode.children, newPath);
+    });
+  };
+
+  if (sharedTree) {
+    traveler([sharedTree], []);
+  }
+  return path;
+}
+
+const PublicBreadcrumb = ({
+  documentId,
+  shareId,
+  sharedTree,
+  children,
+}: Props) => {
+  const path = React.useMemo(
+    () =>
+      pathToDocument(sharedTree, documentId)
+        .slice(0, -1)
+        .map((item) => {
+          return {
+            ...item,
+            url: `/share/${shareId}${item.url}`,
+          };
+        }),
+    [sharedTree, shareId, documentId]
+  );
+  const isNestedDocument = path.length > 2;
```

Comment:
I think this naming is confusing me; it's a nested document as long as firstPath and lastPath are not the same right? This is like... isNestedMoreThanOnce?

I think it might be too hard to explain in a variable name, might be more self-documenting to simply have a case for `path.length === 2` and `path.length > 2` in the return statement.

Reply:
I went ahead and completely refactored the Breadcrumb component 💪